### PR TITLE
[cmake] remove failing lines

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,16 +22,6 @@ cmake_minimum_required(VERSION 2.8)
...
 
-
-set(CMAKE_Fortran_PREPROCESS_SOURCE
-    "<CMAKE_Fortran_COMPILER> -cpp <DEFINES> <INCLUDES> <FLAGS> -E <SOURCE> -o <PREPROCESSED_SOURCE>")
-
-# If we are not building as a part of LLVM, build Flang as an
-# standalone project, using LLVM as an external library:
-if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR )
-  project(Flang)
-endif()
```
Comment (isuruf) [inline CMakeLists.txt:33]: Why is this removed? When this is removed, the 2 lines below will not work

Comment (isuruf) [reply CMakeLists.txt:33]: IMO, last 3 PRs should be reversed

Comment (ghost) [reply CMakeLists.txt:33]: See diff here: https://github.com/flang-compiler/flang/pull/288/files#diff-af3b638bc2a3e6c650974192a53c7291

**Comment by ghost** [comment]
@isuruf Discuss here so that the comments display instantly.

**Comment by ghost** [comment]
To answer your question, the Cmake file had diverged from upstream.

**Comment by isuruf** [comment]
Yes, using `uname -m` is not the most portable way to detect the platform. CMake can do it portably.
They have this comment
```
# The cmake documentation states that these are set.  They are not so we 
 # set them here
```
They are set only when `project(Flang)` is called. So, the correct way is to call `project(Flang)` first

**Comment by ghost** [comment]
If we put this at the top of the file, then the problem should be fixed. It will also keep the diff clear.

```
# If we are not building as a part of LLVM, build Flang as an
 -# standalone project, using LLVM as an external library:
 -if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR )
 -  project(Flang)
 -endif()
```

**Comment by ghost** [comment]
Since you're here, I've been trying to rebuild flang, but have run into inscrutable errors:

https://ci.appveyor.com/project/conda-forge/flang-feedstock/build/1.0.24

**Comment by isuruf** [comment]
That should go away when revert the last 3 PRs

**Comment by isuruf** [comment]
> If we put this at the top of the file, then the problem should be fixed. It will also keep the diff clear.

Isn't that how it was before?
